### DISCUSSION TITLE
HomeAssistantOS-Frigate: PiP in Android

### DISCUSSION THREAD

**Discussion by albotroz** [blakeblackshear/frigate#9251] [Ideas]
Unfortunately, I cannot find anywhere the possibility to minimize a camera on Android (runs via go2rtc) with the help of PiP (Picture in Picture).
I don't want to advertise, but I mean a PiP function like the one in the Android app from Tinycam.
There you can select a camera and select "PiP", then the camera is minimized to a corner. You can also set Tinycam so that the smartphone screen stays on as long as the camera remains on (despite PiP) and with full brightness.

- PiP
- Screen always on
- Full Brightness

Is this possible with HomeAssistantOS-Frigate? I see that PiP is possible on Desktop.

**Comment by NickM-27** [comment]
I use a service called pipup to display on android TV. For Android itself you just start a live view and then press pip, not sure of a more automated way but maybe tasker could be used.

**Reply by albotroz** [reply]
I've just had another look. On Android smartphones, PiP is not possible for my livestreams. Neither via Frigate itself, nor via HomeAssistant.

Normally, this small typical PiP symbol should be displayed for e.g. Go2RTC streams.

I can't say how it is with AndroidTV.

I'll test it more closely tomorrow.

**Reply by NickM-27** [reply]
Are you using go2rtc? It should work without issues on the mse and webrtc views.

**Reply by albotroz** [reply]
Yes go2rtc. Here is my entire config of Frigate and Frigate HASS Cards.
https://github.com/blakeblackshear/frigate-hass-integration/issues/464#issuecomment-1881004299

I don't know if it has something to do with my Frigate HASS Cards settings that no PiP icon is displayed on my Android smartphone when I watch livestreams.

Thanks Nick for your support.

**Reply by NickM-27** [reply]
Oh, I was talking about the frigate ui not frigate hass card.

**Reply by albotroz** [reply]
Smartphone ->
Mh. In Frigate UI there is no PiP icon too. In Frigate UI in webrtc and mse there is only an icon with lines in the top right corner, which is briefly displayed when the stream is full screen.
I have no idea what this is for, because nothing happens when you click on it.

But I mainly want to use the livestreams via Frigate Hass Card.

If you have time and desire you can have a look at my configs.

**Reply by NickM-27** [reply]
I think the issue may be your device then, there is the 3 dots and that can be pressed to run picture in picture
![Screen Shot 2024-01-08 at 07 52 28 AM](https://github.com/blakeblackshear/frigate/assets/14866235/8f51e0df-fd94-4c4c-9fef-cebfed3fb50c)

**Reply by albotroz** [reply]
LineageOS 20, Google Pixel 6a

I have the feeling that there is something wrong between LineageOS 20 and HomeAssistant.

This this typical window for "Picture in Picture" setting is not displayed.

Clicking around touch combinations has minimized viewed livestream from full screen to PiP video, but the display is unusable.
Because most of the live stream is hidden or reduced in size due to the home bar.
![bb](https://github.com/blakeblackshear/frigate/assets/67508404/5ca74aab-c38e-494c-97ec-b5360daf1689)

**Comment by xehn** [comment]
@albotroz I'm not sure if it helps with your problem, but you mentioned Tinycam in your OP, and I wanted to make sure you knew that Tinycam works just fine in parallel with Frigate/go2rtc and Home Assistant.  

I have go2rtc restream active in config, and just gave tinycam on my android devices the go2rtc urls (instead of the direct to camera url).  This allows me to use tinycam as my frontend on my phone, but still have frigate on the backend.  

So to answer the question in your OP - Is this possible?  Yes it is, I use tinycam.

Forgive me if I have misunderstood your issue and you were trying to avoid tinycam for some reason.

**Reply by albotroz** [reply]
Yes, at the moment I mainly continue to use Tinycam and iSpy because I just can't get any further or struggle with Frigate/HomeAssistant configs with some things that are important to me.

**Comment by froman753** [comment]
Any updates on this? I'm also running into the same issue with the HASS card. The PIP Android mobile worked one time for me but I haven't been able to recreate it. Not sure if it was using an alternate stream than WebRTC. I believe I wasn't on local LAN at the time so that may have been it. It may not work with WebRTC streams.

Edit: on a Pixel 6 Pro. I was just messing around with it more. If I hold swiping up from the home pill to bring it to my recent apps, and then swipe up again to go home, it then works to go to PIP. However, going to the recent apps screen turns off the audio and the PIP does not show the audio button to turn it back on.

**Reply by albotroz** [reply]
> Any updates on this? I'm also running into the same issue with the HASS card. The PIP Android mobile worked one time for me but I haven't been able to recreate it. Not sure if it was using an alternate stream than WebRTC. I believe I wasn't on local LAN at the time so that may have been it. It may not work with WebRTC streams.
> 
> Edit: on a Pixel 6 Pro. I was just messing around with it more. If I hold swiping up from the home pill to bring it to my recent apps, and then swipe up again to go home, it then works to go to PIP. However, going to the recent apps screen turns off the audio and the PIP does not show the audio button to turn it back on.

If I set a livestream to full screen and then slide it up ("Home" function) with my finger, a PiP window of the current livestream appears and I can then watch other things such as YouTube while the small PiP window of the livestream remains.
I have 2 livestreams.

However, I have noticed that if I turn off Display and then unlock it again or if HA or Frigate loses the connection to the current livestream or reconnects or updates current Livestream (or whatever), then the PiP window from my other surveillance camera shows Livestream instead of Livestream which I had in my PiP-Window before and with the HA menu bar (see my picture above what I mean with "HA menu bar")

So it doesn't really work properly. Don't how to explain it in a good way.
